lisa/nic.py

```diff
@@ -285,37 +266,33 @@ def reload(self) -> None:
         self._initialize()
 
     @retry(tries=15, delay=3, backoff=1.15)
-    def wait_for_sriov_enabled(self) -> None:
-        lspci = self._node.tools[Lspci]
-
-        # check for VFs on the guest
-        vfs = lspci.get_devices_by_type(constants.DEVICE_TYPE_SRIOV, force_run=True)
-        assert_that(len(vfs)).described_as(
-            "Could not identify any SRIOV NICs on the test node."
-        ).is_not_zero()
-
-        # check if the NIC driver has finished setting up the
-        # failsafe pair, reload if not
-        if not self.get_lower_nics():
-            self.reload()
-        if not self.get_lower_nics():
-            assert_that(self.get_lower_nics()).described_as(
-                "Did not detect any upper/lower sriov paired nics!: "
-                f"upper: {self.get_upper_nics()} "
-                f"lower: {self.get_lower_nics()} "
-                f"unpaired: {self.get_unpaired_devices()}"
-                f"vfs: {','.join([str(pci) for pci in vfs])}"
-            ).is_not_empty()
+    def check_pci_enabled(self, pci_enabled: bool) -> None:
+        self.reload()
+        if pci_enabled:
+            assert_that(len(self.get_device_slots())).described_as(
+                "Could not identify any pci devices on the test node."
+            ).is_not_zero()
+            if self._is_mana_device_discovered() and not self._node.tools[
+                KernelConfig
+            ].is_enabled("CONFIG_MICROSOFT_MANA"):
+                return
+            assert_that(pci_enabled).described_as(
+                "AN enablement and pci device are inconsistent"
+            ).is_equal_to(any(self.get_pci_nics()))
+        else:
+            assert_that(self.get_device_slots()).described_as(
+                "pci devices still on the test node."
+            ).is_empty()
 
     def _initialize(self, *args: Any, **kwargs: Any) -> None:
         self._node.log.debug("loading nic information...")
-        self.nic_names = self._get_nic_names()
-        self._get_node_nic_info()
-        self._get_default_nic()
-        self.load_interface_info()
+        self._nic_names = self._get_nic_names()
+        self._load_nics()
+        self.nics_list: List[str] = list(self.nics.keys())
```

Comment:
Is it the same as `_nic_names`? If so, should the `_get_nic_names` be removed?

Reply:
I check back to us get_nic_names method